Extractor/Config/SourceConfig.cs

```diff
@@ -315,6 +315,38 @@ public HashSet<uint> FinalRetryStatusCodes
                 return finalRetryStatusCodes;
             }
         }
+
+        public bool ShouldRetryException(Exception ex, IEnumerable<uint> statusCodes)
+        {
+            if (ex is ServiceResultException serviceExc)
+            {
+                var code = serviceExc.StatusCode;
+                return statusCodes.Contains(serviceExc.StatusCode);
+            }
+            else if (ex is ServiceCallFailureException failureExc)
+            {
+                return failureExc.Cause == ServiceCallFailure.SessionMissing;
+            }
+            else if (ex is SilentServiceException silentExc)
+            {
+                if (silentExc.InnerServiceException != null)
+                {
+                    return statusCodes.Contains(silentExc.InnerServiceException.StatusCode);
+                }
+            }
+            else if (ex is AggregateException aex)
+            {
+                // Only retry aggregate exceptions if one of the inner exceptions should be retried...
+                var flat = aex.Flatten();
+                return aex.InnerExceptions.Any(e => ShouldRetryException(e, statusCodes));
```

Comment:
Should we not use the flattened exception here? Or remove it?

Reply:
Yep, typo.